Tareas/Tercer Tarea/DanielNavarrete/NewCodes/Follow/FollowObject.cs

```diff
@@ -0,0 +1,102 @@
+using UnityEngine;
+
+public class FollowObject : SteeringBehaviors
+{
+    [Header("Follow Dependencies")]
+    public GameObject leader;
+    public SteeringBehaviors leaderBehaviors;
+
+    [HideInInspector]
+    public bool followEnable = false;
+    [HideInInspector]
+    public bool enableFlee = false;
+    [HideInInspector]
+    public bool enablePursuit = false;
+
+    [Header("Speed Intervals")]
+    SpeedIntervals speedIntervals;
+    public bool enableInterval = false;
+
+    private SpriteRenderer sprite;
+
+    //Avoid Walls
+    AvoidWalls avoidWalls;
+
+    void Start()
+    {
+        leader = GameObject.FindGameObjectWithTag("Player");
+        leaderBehaviors = leader.GetComponent<SteeringBehaviors>();
+        speedIntervals = GetComponent<SpeedIntervals>();
+        avoidWalls = GetComponentInChildren<AvoidWalls>();
+        sprite = GetComponentInChildren<SpriteRenderer>();
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if(!enablePursuit)
+        {
+            FollowLeader();
+            FleePlayer();
+        }
+        else if(enablePursuit)
+        {
+            PursuitPlayer();
+        } 
+    }
+
+    private void OnTriggerEnter2D(Collider2D collision)
+    {
+        if(collision.CompareTag("Player"))
+        {
+            followEnable = true;
+        }
+        if(collision.CompareTag("Wall"))
+        {
+            transform.position = Vector3.zero;
+        }
+    }
+
+    void FollowLeader()
+    {
+        if (followEnable) transform.position += Seek(Follow(leaderBehaviors, leader)) * Time.deltaTime;
+        else if (!followEnable) this.gameObject.GetComponent<CircleCollider2D>().enabled = true;
+    }
+
+    void FleePlayer()
+    {
+        if(!followEnable && enableFlee)
+        {
+            if (speedIntervals != null && enableInterval) this.speed = speedIntervals.SpeedInterval(leader.transform.position);
+            else this.speed = 0;
```

Comment:
Te comente que el else lo quitaras, es un paso extra.

Pon el this.speed antes del if y luego el if, 
despues no lo necesitaras.
-0.5 puntos

Reply:
Listo